api/error.go

```diff
@@ -45,14 +45,14 @@ func (e Error) LogData() map[string]interface{} {
 func statusCode(err error) int {
 	var serr coder
 	switch {
-	case errors.As(err, &serr):
-		return serr.Code()
 	case dperrors.Unavailable(err):
 		return http.StatusServiceUnavailable
 	case dperrors.NotFound(err):
 		return http.StatusNotFound
 	case dperrors.Conflict(err):
 		return http.StatusConflict
+	case errors.As(err, &serr):
```

Comment:
Why did you move this? This was purposely first so it would be evaluated first and take precedence

Reply:
with the previous order the error return was always a 500 and never a 404, based on the error code returned by the DAO/MongoDB pakage.

Reply:
This has been addressed.